hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -168,21 +167,21 @@ public class HoodieWriteConfig extends HoodieConfig {
 
   public static final ConfigProperty<String> WRITE_PAYLOAD_CLASS_NAME = ConfigProperty
       .key("hoodie.datasource.write.payload.class")
-      .defaultValue(DefaultHoodieRecordPayload.class.getName())
+      .noDefaultValue()
       .markAdvanced()
+      .deprecatedAfter("1.0.0")
```

Comment:
I see your point that payload class is no longer required in Hudi 1.0 tables so that no default should be set.  For Hudi 1.0 release reading 0.x tables, should we still maintain a variable else where to track the default payload class to be used?  The backwards compatibility read logic can be handled in a follow-up.

Reply:
Created static var in the tableconfig to track the default and replaced the uses where we hardcoded the default payload